src/ng/sniffer.js

```diff
@@ -21,8 +21,13 @@ function $SnifferProvider() {
   this.$get = ['$window', '$document', function($window, $document) {
     var eventSupport = {},
         // Chrome Packaged Apps are not allowed to access `history.pushState`. They can be detected by
-        // the presence of `chrome.app.runtime` (see https://developer.chrome.com/apps/api_index)
-        isChromePackagedApp = $window.chrome && $window.chrome.app && $window.chrome.app.runtime,
+        // the presence of `chrome.app.runtime` (see https://developer.chrome.com/apps/api_index).
+        // For sandboxed apps, check for an extension runtime ID, but no access to other Chrome
+        // runtime APIs. See https://developer.chrome.com/apps/manifest/sandbox
+        isChromePackagedApp =
+            $window.chrome &&
+            ($window.chrome.app && $window.chrome.app.runtime ||
+                !$window.chrome.app && $window.chrome.runtime && $window.chrome.runtime.id),
```

Comment:
ah, I didn't pay much attention to the `!$window.chrome.app` check. It should be good then.
I wish there were a more straight-forward way to detect chrome packaged apps :disappointed: